soufi/finders/yum.py

```diff
@@ -137,13 +136,13 @@ def _nevra_or_none(self, package):
         return nevra['name'], f"{nevra['ver']}-{nevra['rel']}"
 
     # Cache repo downloads as they are slow and network-bound.
-    @classmethod
-    @functools.lru_cache(maxsize=1024)
-    def _get_repo(cls, url):
+    def _get_repo(self, url):
         if not url.endswith('/'):
             url += '/'
         try:
-            return repomd.load(url)
+            return self._cache.get_or_create(
+                f"repo-{url}", repomd.load, creator_args=([url], {})
+            )
```

Comment:
How unfortunate: these objects are effectively `ElementTree` objects, which very much do not like to be serialized, so this will not work with e.g., `dogpile.cache.memorypickle` or `dogpile.cache.redis`.

The `repomd` library went with simplicity over configurability, so we'll have to write our own `load` method if we want to cache the inputs instead.  Either way, this should not merge as-is.